### PR TITLE
Support normalising household consumption to adult female equivalent supply

### PR DESCRIPTION
**Pull request by bgsandan**

### CODE REVIEW

File: db/objects/household/V0.1.0.45__household_factor.sql [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+COMMENT on column household_normalisation_factor.energy_requirement_m IS 'Energy requirement for a male of this age';
+COMMENT on column household_normalisation_factor.energy_requirement_f IS 'Energy requiement for a female of this age';
+COMMENT on column household_normalisation_factor.ame_m                IS 'Adult Male Equivalent (AME) adjustment factor for a male of this age';
+COMMENT on column household_normalisation_factor.ame_f                IS 'Adult Male Equivalent (AME) adjustment factor for a female of this age';
+COMMENT on column household_normalisation_factor.afe_m                IS 'Adult Female Equivalent (AFE) adjustment factor for a male of this age';
+COMMENT on column household_normalisation_factor.afe_f                IS 'Adult Female Equivalent (AFE) adjustment factor for a female of this age';
+
+
+
+INSERT INTO household_normalisation_factor(age_in_months,energy_requirement_m,energy_requirement_f,ame_m,ame_f,afe_m,afe_f) VALUES (0,0,0,0,0,0,0);
```
Comment (rbroth) [inline db/objects/household/V0.1.0.45__household_factor.sql:22]: I'd say break out the the normalisation factors data into a separate file. Are they liable to change between surveys/studies? Or is this something that is well-known, widely used and unchanging? I.e. Are they more like biomarker thresholds or more like the list of ISO country codes? If the former, perhaps put it into the data repo, if the latter, put it into the dictionary data folder

File: db/objects/household/R__018_household_factor.sql [hunk 2]
```diff
@@ -0,0 +1,10 @@
+create or replace view household_normalisation as
+
+with factors as (
```
Comment (rbroth) [inline db/objects/household/R__018_household_factor.sql:3]: formatting: newline before FROM, WHERE, GROUP BY, etc

**Review by rbroth — APPROVED** [review]

File: db/objects/household/R__018_household_factor.sql (hunk 2 above)
Comment (rbroth) [inline db/objects/household/R__018_household_factor.sql:3]: Also, a more stylistic choice: You use WITH a lot; I'd probably use subqeries instead, might make it easier to read, so:

```sql
create or replace view household_normalisation as


select household_id, sum(afe_factor) as afe_factor, sum(ame_factor) as ame_factor, sum(count) as member_count 
from  
(
    -- AFE/AME for females
    select hm.household_id, sum(hnf.afe_f) as afe_factor, sum(hnf.ame_f) as ame_factor, count(hm.id) 
    from household_member hm 
    join household_normalisation_factor hnf on hm.age_in_months=hnf.age_in_months 
    where hm.sex='f' 
    group by hm.household_id, hm.sex  
)
union
(
    -- AFE/AME for males
    select hm.household_id, sum(hnf.afe_m) as afe_factor, sum(hnf.ame_f) as ame_factor, count(hm.id) 
    from household_member hm 
    join household_normalisation_factor hnf on hm.age_in_months=hnf.age_in_months 
    where hm.sex='m' 
    group by hm.household_id, hm.sex  
)
group by household_id
```

File: db/objects/household/R__025_household_intake_normalised.sql [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        , (hi.Biotin_in_mcg/hn.afe_factor              ) as  Biotin_in_mcg
+        , (hi.VitaminC_in_mg/hn.afe_factor             ) as  VitaminC_in_mg
+        , (hi.VitaminD_in_mcg/hn.afe_factor            ) as  VitaminD_in_mcg
+        , (hi.VitaminE_in_mg/hn.afe_factor             ) as  VitaminE_in_mg
+        , (hi.PhyticAcid_in_mg/hn.afe_factor           ) as  PhyticAcid_in_mg
+
+
+from household_intake hi join household_normalisation hn on hi.household_id=hn.household_id;
+
+COMMENT ON VIEW household_intake_normalised IS 'View of amount of micronutrients consumed in total by individual households normalised to AFE';
```
Comment (rbroth) [inline db/objects/household/R__025_household_intake_normalised.sql:50]: Expand acronym in comment to "Adult Female Equivalent (AFE)"

File: db/objects/household/R__019_afe_ear_threshold.sql [hunk 4]
```diff
@@ -0,0 +1,18 @@
+create or replace view afe_ear_threshold as
```
Comment (rbroth) [inline db/objects/household/R__019_afe_ear_threshold.sql:1]: Add comment describing the view

File: db/objects/household/R__018_household_factor.sql [hunk 5]
```diff
@@ -0,0 +1,41 @@
+create
+or replace view household_normalisation as with factors as (
```
Comment (rbroth) [inline db/objects/household/R__018_household_factor.sql:2]: Add comment describing the view

File: db/objects/household/R__026_household_intake_pivot.sql [hunk 6]
```diff
@@ -0,0 +1,59 @@
+create
+or replace view household_intake_deficiency_pivot as
```
Comment (rbroth) [inline db/objects/household/R__026_household_intake_pivot.sql:2]: Add comment describing the view

File: db/objects/household/V0.1.0.45__household_factor.sql [hunk 7]
```diff
@@ -0,0 +1,18 @@
+CREATE TABLE household_normalisation_factor (
```
Comment (rbroth) [inline db/objects/household/V0.1.0.45__household_factor.sql:1]: Might it be worthwhile to add a column "sex" and thus get rid of duplicate columns?

Comment (bgsandan) [reply db/objects/household/V0.1.0.45__household_factor.sql:1]: The current structure reflects the way the source data is structured.  Maybe one for the refactor backlog as at least it works as is?

**Comment by bgsandan** [comment]
Comments added to views :memo:

File: db/objects/household/R__018_household_factor.sql [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
```
Comment (rbroth) [inline db/objects/household/R__018_household_factor.sql:2]: newline before WITH. Also, would be nice to capitalize keywords

**Comment by rbroth** [comment]
These sets of tables are normalzied to AFE; will we have similar tables and views normalised to AME? IF so, might be worth looking at combing them and having a discriminator column instead

File: db/objects/household/R__025_household_intake_normalised.sql [hunk 9]
```diff
@@ -0,0 +1,50 @@
+create or replace view household_intake_normalised as
```
Comment (rbroth) [inline db/objects/household/R__025_household_intake_normalised.sql:1]: name should indicate that this is for AFE, especially if we'll have similar views for AME

File: db/objects/household/R__026_household_intake_pivot.sql [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+select
+    household_supply.*,
+    afe_ear_threshold.afe_ear,
+    (
+        household_supply.micronutrient_supply < afe_ear_threshold.afe_ear
+    ) as deficient
```
Comment (rbroth) [inline db/objects/household/R__026_household_intake_pivot.sql:8]: boolean columns should be prefixed with is_ i.e. is_deficient

File: db/objects/household/V0.1.0.45__household_factor.sql (hunk 7 above)
Comment (rbroth) [reply db/objects/household/V0.1.0.45__household_factor.sql:1]: Sure, add an issue

Comment (bgsandan) [reply db/objects/household/V0.1.0.45__household_factor.sql:1]: Changes made as suggested

**Pull request merged**